opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -90,20 +94,31 @@ where
 
 impl<S, P, L> Layer<S> for OpenTelemetryTracingBridge<P, L>
 where
-    S: tracing::Subscriber,
+    S: Subscriber + for<'a> LookupSpan<'a>,
     P: LoggerProvider<Logger = L> + Send + Sync + 'static,
     L: Logger + Send + Sync + 'static,
 {
-    fn on_event(
-        &self,
-        event: &tracing::Event<'_>,
-        _ctx: tracing_subscriber::layer::Context<'_, S>,
-    ) {
+    fn on_event(&self, event: &tracing::Event<'_>, ctx: tracing_subscriber::layer::Context<'_, S>) {
         let meta = event.metadata();
         let mut log_record: LogRecord = LogRecord::default();
         log_record.severity_number = Some(severity_of_level(meta.level()));
         log_record.severity_text = Some(meta.level().to_string().into());
 
+        // Extract the trace_id & span_id from the opentelemetry extension.
+        if let Some((trace_id, span_id)) = ctx.event_span(event).and_then(|span| {
```

Comment:
just curious, can't we get the currently active span using `ctx.lookup_current()`. How is `ctx.event_span(event)` different ?

Reply:
Yes I believe we can, updating!